qiskit/compiler/assembler.py

```diff
@@ -243,24 +291,124 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
                 RuntimeWarning,
             )
 
+    qubit_lo_freq = qubit_lo_freq or getattr(backend_defaults, 'qubit_freq_est', None)
+    meas_lo_freq = meas_lo_freq or getattr(backend_defaults, 'meas_freq_est', None)
+
+    qubit_lo_range = qubit_lo_range or getattr(backend_config, 'qubit_lo_range', None)
+    meas_lo_range = meas_lo_range or getattr(backend_config, 'meas_lo_range', None)
+
+    # check lo frequencies
+    _check_lo_freqs(qubit_lo_freq, meas_lo_freq, qubit_lo_range, meas_lo_range, n_qubits)
+
     # create run configuration and populate
     run_config_dict = dict(shots=shots,
                            memory=memory,
                            max_credits=max_credits,
                            seed_simulator=seed_simulator,
                            init_qubits=init_qubits,
                            rep_delay=rep_delay,
+                           qubit_lo_freq=qubit_lo_freq,
+                           meas_lo_freq=meas_lo_freq,
+                           qubit_lo_range=qubit_lo_range,
+                           meas_lo_range=meas_lo_range,
                            **run_config)
 
     return qobj_id, qobj_header, run_config_dict
 
 
-def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
-                      meas_lo_range, schedule_los, meas_level,
-                      meas_return, meas_map,
-                      memory_slot_size,
-                      rep_time, parametric_pulses,
-                      **run_config):
+def _check_lo_freqs(
+    qubit_lo_freq: Union[None, List[float]],
+    meas_lo_freq: Union[None, List[float]],
+    qubit_lo_range: Union[None, List[List[float]]],
+    meas_lo_range: Union[None, List[List[float]]],
+    n_qubits: Union[None, int],
+):
+    """Check lo frequency arrays are the correct size and in range.
+
+    Args:
+        qubit_lo_freq: List of qubit lo frequencies.
+        meas_lo_freq: List of meas lo frequencies.
+        qubit_lo_range: List of allowed qubit lo frequencies. Each element is 2d list of form
+            ``[min_qubit_lo_freq, max_qubit_lo_freq]``.
+        meas_lo_range: List of allowed meas lo frequencies. Each element is 2d list of form
+            ``[min_meas_lo_freq, max_meas_lo_freq]``.
+
+    """
+    # only check if is a valid number of qubits in system
+    if not n_qubits:
+        return
```

Comment:
What is the purpose of this? Does it just check if n_qubits is greater than 0? Is this necessary?

Reply:
`n_qubits` could be `None`. See line 240

Reply:
I think it better to check this when it may occur than to have the delayed check down here.

Reply:
Addressed in https://github.com/Qiskit/qiskit-terra/pull/6167/commits/705ca3c97ae84c6f950f430f1eea576d4eb93e1c.